### ISSUE TITLE
Migration tools and/or guides for 1Password

### ISSUE THREAD

**Issue by vit9696** [keepassium/KeePassium#177]
Currently I use 1Password to store my credentials. Due to constant sync issues and their new policy forcing me to pay on a subscription basis, I am looking for an alternative. This alternative can be KeePass with KeePassium as a client on both macOS and iOS.

It will help a lot to provide a decent manual on how to perform such a migration without data loss. For example, I have a database with almost a thousand of categorised entries: login + password pairs, passwords, notes, software licenses, cards. A fair amount of them has custom fields.

Providing a GUI solution may take a lot of effort, but at least having some tool and a step-by-step guide may bring some clients, who are currently looking for a solution.

**Comment by keepassium** [comment]
This is a very good point! I was going to write a guide _one day_, but your message pushed me to actually start. 

I got a proof-of-the-concept outline of the necessary steps. It will take some time to refine it and make screenshots, though. 

Good news: it seems KeePassXC can handle all the steps, no extra tools required :)

**Comment by rcarmo** [comment]
I can confirm KeePassXC imports 1Password vaults without any issues (that I've noticed so far).

**Comment by vit9696** [comment]
Indeed, KeePassXC did the trick for me, although it took me some time to figure out how to properly export the database in a compatible format. The issues I spotted so far:

- I had unnamed fields and it warned me about them (`Some entries have custom field(s) with empty names. This can be a sign of data corruption. Please check these entries:`)
- Multiple URLs are not supported by KeePassium (KP2A_URL_1, KP2A_URL_2, etc.)
- Application icons and website icons are permanently lost

**Comment by keepassium** [comment]
@vit9696 , I am not sure whether this is a "Document" file type:
<img width="1079" alt="Screen Shot 2021-10-09 at 16 31 12" src="https://user-images.githubusercontent.com/38399598/136662338-b352ea99-20d3-4dfa-8223-aa8b72c9ad6e.png">

The selected item did not make it to the local vault:

<img width="1079" alt="Screen Shot 2021-10-09 at 16 31 15" src="https://user-images.githubusercontent.com/38399598/136662429-08cb8746-044f-41c8-ac73-8f870987822a.png">

**Comment by keepassium** [comment]
> Hmm, `local` vault? I do not use 1password.com, it is just the local vault. Maybe this is a difference.

Yep, I think this explains the difference. Half of the guide is about saving the cloud data to a tangible `.opvault` file, and this was the lossy step. I did not try re-adding attachments to the local vault, because it's an intermediate step anyway…

**Comment by vit9696** [comment]
I guess it sounds like a bug in 1Password itself then. Maybe leave a note to skip these steps if one is already using a local vault and that there is no issue with local vaults…

**Comment by keepassium** [comment]
> Maybe leave a note to skip these steps if one is already using a local vault and that there is no issue with local vaults…

Good point, done. Thanks!
